app/code/Magento/Inventory/Ui/Component/MassAction/Filter.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Ui\Component\MassAction;
+
+use Magento\Ui\Component\MassAction\Filter as BaseFilter;
+use Magento\Framework\Api\Search\DocumentInterface;
+
+/**
+ * Class Filter
+ * @todo remove this class after code from Engine\MagentoFix\Ui\Component\MassAction\Filter would be deploy to
+ *       production
+ */
+class Filter
+{
+    /**
+     * @var BaseFilter
+     */
+    private $filter;
+
+    /**
+     * @param BaseFilter $filter
+     */
+    public function __construct(
+        BaseFilter $filter
+    ) {
+        $this->filter = $filter;
+    }
+
+    /**
+     * Get ids from search filter
+     *
+     * @return array
+     */
+    public function getIds()
+    {
+        $this->filter->applySelectionOnTargetProvider();
+        $component = $this->filter->getComponent();
+        $dataProvider = $component->getContext()->getDataProvider();
+        $searchResult = $dataProvider->getSearchResult();
```

Comment:
I see a violation of encapsulation and Law of Demeter here.
Because current code is equivalent to 
`$this->filter->getComponent()->getContext()->getDataProvider()->getSearchResult()`

@naydav is it really recommended way of applying Filtering?

Reply:
GitHub issue number https://github.com/magento/magento2/issues/10988